scripts/ci-local.sh

```
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+# Run the same gates as .github/workflows/ci.yml locally, in the same order,
+# with the same RUSTFLAGS. Mirrors each job 1:1 so a green run here predicts a
+# green CI. Pass --fast to skip the heavy full-workspace + local-feature builds.
+#
+#   scripts/ci-local.sh         # everything CI runs
+#   scripts/ci-local.sh --fast  # skip --all-features / --features local builds
+set -uo pipefail
+
+export RUSTFLAGS="${RUSTFLAGS:--D warnings}"   # CI sets this globally (ci.yml env)
+FAST=0; [ "${1:-}" = "--fast" ] && FAST=1
+ROOT="$(cd "$(dirname "$0")/.." && pwd)"; cd "$ROOT"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)